transmogrifier/sources/json/aardvark.py

```diff
@@ -0,0 +1,121 @@
+import logging
+
+import transmogrifier.models as timdex
+from transmogrifier.sources.transformer import JsonTransformer
+
+logger = logging.getLogger(__name__)
+
+
+class Aardvark(JsonTransformer):
+    """Aardvark transformer."""
+
+    @classmethod
+    def get_main_titles(cls, source_record: dict) -> list[str]:
+        """
+        Retrieve main title(s) from a Aardvark JSON record.
+
+        Overrides metaclass get_main_titles() method.
+
+        Args:
+            source_record: A JSON object representing a source record.
+        """
+        titles = []
+        if title := "dct_title_s" in source_record and source_record["dct_title_s"]:
+            titles.append(title)
+        return titles
+
+    @classmethod
+    def get_source_record_id(cls, source_record: dict) -> str:
+        """
+        Get source record ID from a JSON record.
+
+        Args:
+            source_record: A JSON object representing a source record.
+        """
+        return source_record["id"]
+
+    @classmethod
+    def record_is_deleted(cls, source_record: dict) -> bool:
+        """
+        Determine whether record has a status of deleted.
+
+        ## WIP - defining to enable instantiation of Aardvark instance.
+
+        Args:
+            source_record: A JSON object representing a source record.
+        """
+        return False
+
+    def get_optional_fields(self, source_record: dict) -> dict | None:
+        """
+        Retrieve optional TIMDEX fields from a Aardvar JSON record.
+
+        Overrides metaclass get_optional_fields() method.
+
+        Args:
+            xml: A BeautifulSoup Tag representing a single Datacite record in
+                oai_datacite XML.
+        """
+        fields: dict = {}
+
+        # alternate_titles field not used in Aardvark
+
+        # content_type
+        fields["content_type"] = ["Geospatial data"]
+
+        # contributors
+
```

Comment:
Should there be a function for every optional MITAardvark fields that have a mapping to a Timdex field (e.g. `contributors`) _at this point_?: <img width="1038" alt="image" src="https://github.com/MITLibraries/transmogrifier/assets/29102054/b5cea8f0-17cf-48f9-89cf-f0e6a23ef524">

Or is that work for a later PR/ticket? 

If the latter, thanks for clarifying!

Reply:
Yes, this is just the start, other methods coming a few at a time in subsequent PRs